File: 1100/1829E/main.cpp
#include<bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin>>t;

    while(t--) 
    {
        int n, m;
        cin>>n>>m;

        vector<vector<int>> a(n, vector<int>(m));
        for(int i=0; i<n; i++) 
        {
            for(int j=0; j<m; j++)
                cin>>a[i][j];
        }

        int cnt = 0;
        int dr[] = {-1, 1, 0, 0};
        int dc[] = {0, 0, -1, 1};

        for(int i=0; i<n; i++) 
        {
            for(int j=0; j<m; j++) 
            {
                if(a[i][j] == 0)
                    continue;

                int c = 0;
                queue<pair<int, int>> q;
                q.push({i, j});

                while(!q.empty()) 
                {
                    int c_i = q.front().first;
                    int c_j = q.front().second;

                    q.pop();

                    if(a[c_i][c_j] == 0)
                        continue;

                    c += a[c_i][c_j];
                    a[c_i][c_j] = 0; // Mark as visited

                    for(int d=0; d<4; d++) 
                    {
                        int ni = c_i + dr[d];
                        int nj = c_j + dc[d];

                        if(ni >= 0 && ni < n && nj >= 0 && nj < m && a[ni][nj] != 0)
                            q.push({ni, nj});
                    }
                }

                cnt = max(cnt, c);
            }
        }

        cout<<cnt<<"\n";
    }

    return 0;
}

File: 1100/1829E/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
 
        while(t-- > 0)
        {
            int n = sc.nextInt();
            int m = sc.nextInt();

            // Input the matrix
            int a[][] = new int[n][m];
            for(int i=0;i<n;i++)
            {
                for(int j=0;j<m;j++)
                    a[i][j] = sc.nextInt();
            }
 
            int cnt = 0;

            // BFS to find connected components
            for(int i=0;i<n;i++)
            {
                for(int j=0;j<m;j++)
                {
                    if(a[i][j] == 0)
                        continue;

                    // Start BFS
                    int c = 0;
                    Queue<int[]> q = new LinkedList<>();
                    q.add(new int[]{i,j});
 
                    while(q.size() > 0)
                    {
                        int idx[] = q.poll();
                        int c_i = idx[0];
                        int c_j = idx[1];

                        // If the cell is already visited or 0
                        // Skip it
                        if(a[c_i][c_j] == 0)
                            continue;
        
                        c += a[c_i][c_j];
                        a[c_i][c_j] = 0; // Mark as visited

                        int s = q.size();

                        // Check all 4 directions
                        if(c_i-1 >= 0 && a[c_i-1][c_j] != 0)
                            q.add(new int[]{c_i-1,c_j});
                        if(c_i+1 < n && a[c_i+1][c_j] != 0)
                            q.add(new int[]{c_i+1,c_j});
                        if(c_j-1 >= 0 && a[c_i][c_j-1] != 0)
                            q.add(new int[]{c_i,c_j-1});
                        if(c_j+1 < m && a[c_i][c_j+1] != 0)
                            q.add(new int[]{c_i,c_j+1});
                    }
 
                    cnt = Math.max(cnt, c);
                }
            }
 
            System.out.println(cnt);
        }
    }
}